dask/dataframe/io/tests/test_parquet.py

```diff
@@ -3424,6 +3424,36 @@ def test_pyarrow_dataset_filter_partitioned(tmpdir, split_row_groups):
     )
 
 
+def test_pyarrow_dataset_filter_on_partitioned(tmpdir, engine):
+    # See: https://github.com/dask/dask/issues/9246
+    ddf = dd.from_map(
+        lambda x: pd.DataFrame({"a": [x], "b": [x]}),
+        "abcdefg",
+    )
+    ddf["b"] = ddf["b"].astype("category")
+    ddf.to_parquet(tmpdir, engine=engine, partition_on=["b"])
+
+    # Check List[Tuple] filters
+    read_ddf = dd.read_parquet(
+        tmpdir,
+        engine=engine,
+        filters=[("b", "==", "c")],
+    )
+    assert read_ddf.npartitions == 1
+    assert_eq(read_ddf["a"], ddf.partitions[2]["a"])
```

Comment:
Is selecting the `"a"` column here meaningful? Or could we just assert that `read_ddf` and `ddf.partitions[2]` are the same?

Reply:
I didn't want to deal with the object vs category column comparison in `"b"`, when the primary test is that we are only getting a single (non-empty) partition.

Reply:
I see. Is the category part needed? Could we drop `ddf["b"] = ddf["b"].astype("category")`, or is that needed for this test?

Reply:
>I see. Is the category part needed? Could we drop ddf["b"] = ddf["b"].astype("category"), or is that needed for this test?

Oops - I meant to remove that line! (I added it before bailing on the direct column-"b" comparison)

Reply:
Removed that line and added some short comments. However, if you think it makes sense, I can also (1) Add special logic to fastparquet to handle `[[Tuple]]` in the fastparquet engine, and/or (2) add some code to enable a "full" `assert_eq` on the result.

Reply:
> Add special logic to fastparquet to handle `[[Tuple]]` in the fastparquet engine

If you've got bandwidth for that, then feel free to (certainly no obligation though). Though probably in a separate PR to keep the scope of this one small

> add some code to enable a "full" assert_eq on the result

That would be nice if doing so is relatively straightforward